File: concurrency/code_samples/Counter.java
public class Counter implements Runnable {

	private int my_num;
	
	public Counter(int my_num){
		this.my_num = my_num;
	}

	public void run(){
		System.out.println("Hello from " + my_num + "!");
	}

	public static void main(String[] args){
		Thread[] threads = new Thread[10];
		for(int i=0; i<10; i++){
			threads[i] = new Thread(new Counter(i));
			threads[i].start();
		}

		for(int i=0; i<10; i++){
			try{
				threads[i].join();
			} catch (Exception ex){

			}
		}
	}

}

File: concurrency/code_samples/counter.c
#include <pthread.h>
#include <stdio.h>
#include <stdlib.h>

typedef struct params {
	int id;
} params_t;

void* say_hi(void* arg){
	int id;
	id = (*(params_t*)(arg)).id;
	printf("Hello from %d\n", id);
}

int main(int argc, char** argv){
	pthread_t threads[10];
	params_t params;
	for(int i=0; i<10; i++){
		params.id = i;
		if(pthread_create(&threads[i], NULL, say_hi, &params)){
			// Thread created.
		}
	}

	for(int i=0; i<10; i++){
		pthread_join(threads[i], NULL);
	}
}
